types/tx/service.go

```diff
@@ -0,0 +1,123 @@
+package tx
```

Comment:
This service implementation shouldn't be in `types/tx`, rather `x/auth/tx`.

Reply:
I actually also tried that 😅, but then baseapp import `x/auth/tx` (because of https://github.com/cosmos/cosmos-sdk/pull/7688#discussion_r513471016), which seems like a no-go (and creates import cycles).

Maybe I can try to refactor https://github.com/cosmos/cosmos-sdk/pull/7688#discussion_r513471016 in this PR?

Reply:
The previous tx endpoints lived in `x/auth` not baseapp, so we could do the same thing and move the baseapp function to `x/auth`. Maybe eventually there's a separate `tx` module because I think they're slightly separate concerns but this should work for now.

Reply:
Also can't this happen in `x/auth` `RegisterServices` or we don't have a `client.Context` there?

Reply:
> Also can't this happen in x/auth RegisterServices or we don't have a client.Context there?

yeah, we don't have a client.Context there, it's called from simapp.

> so we could do the same thing and move the baseapp function to x/auth

I think that could work, I'll try that.

Reply:
Addressed.